setup_win.bat

```diff
@@ -138,35 +139,51 @@ for %%R in (HKLM HKCU) do (
 		rem so we skip the first two lines (skip=2) and then we take the second token
 		rem and the reamining output (tokens=2*), so %%A is REG_SZ and %%B is the path
 		rem even if it contains spaces (tokens are delimited by spaces)
-		echo Trying registry root %%R [%%T]...
+		echo    Trying registry root %%R [%%T]...
 		for /f "usebackq skip=2 tokens=2*" %%A in (`REG QUERY "%%R\SOFTWARE\Inkscape\Inkscape" /ve /reg:%%T 2^>nul`) do (
-			set INKSCAPE_DIR=%%B
+			if exist %%B (
+				set INKSCAPE_DIR=%%B
+			)
 		)
 		if defined INKSCAPE_DIR (
-			echo Inkscape considered to be installed in !INKSCAPE_DIR!
+			echo    Inkscape considered to be installed in !INKSCAPE_DIR!
 			set INKSCAPE_DIR=!INKSCAPE_DIR!\bin
-			echo Setting executable path to !INKSCAPE_DIR!
+			echo    Setting executable path to !INKSCAPE_DIR!
 			set INKSCAPE_EXE=!INKSCAPE_DIR!\inkscape.exe
 			if exist "!INKSCAPE_EXE!" (
 			    echo !INKSCAPE_EXE! found
 				echo.
-				goto INKSCAPE_FOUND
+				goto    INKSCAPE_FOUND
 			) else (
-				echo !INKSCAPE_EXE! not found
+				echo    !INKSCAPE_EXE! not found
 			)
-			rem for %%S in ("!INKSCAPE_EXE!") do set INKSCAPE_DIR=%%~dpS
 		)
 	)
 )
 
+rem If we did non succeed in the registry lets have a look 
+rem at the most common install locations
+echo Trying the usual Windows install locations...
+for %%D in (C, D, E, F, G, H) do (
+	for %%F in ("Program Files", "Program Files (x86)") do (
```

Comment:
I suggest to rename one-letter variables. The bigger code lifetime and variable scope the more chances it should be named better.

Reply:
Unfortunately, in Windows batch only single letter variables are allowed for the iteration variable in for-loops. One of the strange things in this language...

Reply:
Didn't know that... Indeed a surprising feature :confused: